docs(consensus): update iota architecture

NaitsabesMue opened the pull request:
# Description of change

This PR refines various sections of the IOTA documentation to improve clarity and accuracy. Key updates include:

Changes
	•	Quiz Fixes: Updated quiz answers to reflect multiple correct responses where applicable and ensured factual accuracy.
	•	Validator Accountability: Removed unsupported claims regarding active detection mechanisms.
	•	Checkpoint Description: Clarified how transactions are ordered and finalized, avoiding misleading “causal ordering” terminology.
	•	Finality Explanation: Improved wording around transaction finality and clarified that epoch duration is an upper bound for latency.
	•	Intro Wording: Minor refinements to improve readability while maintaining structure.
	•	Bullshark-Narwhal Reference: Removed outdated mentions to align with the current protocol.

Why These Changes?

These modifications ensure the documentation is technically accurate and enhances readability without altering the intended meaning.

## Links to any relevant issues

Closes #5414

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Documentation Fix

## Testing Required

Note: The quiz now includes multiple correct answers for some questions. Please verify that the system correctly supports this behavior before merging.

NaitsabesMue committed:
Update quiz answers to reflect multiple correct responses for asset security and privacy questions; verify multi-answer support.

NaitsabesMue committed:
Clarify validator accountability description: remove unsupported claims of active censorship detection mechanisms.

NaitsabesMue committed:
Refine checkpoint description

NaitsabesMue committed:
Remove outdated Bullshark-Narwhal reference; clarify finality timing and effects certificate usability.

NaitsabesMue committed:
Minor wording tweak to intro: retain structure while clarifying IOTA’s distinction from traditional blockchains

NaitsabesMue commented:
The quiz seems not to support multiple correct answers (at least in the current state).  The easiest would be to change the answers so there is only one clear, accurate answer. Wdyt?

lucas-tortora commented on docs/site/static/json/about-iota/iota-architecture/iota-security.json:
```diff
@@ -30,16 +30,16 @@
         "questionText": "Considering the public auditability of all transactions on IOTA, which of the following could be a potential vulnerability for users seeking privacy, even if they use pseudonymous addresses?",
         "answerOptions": [
-          { "answerText": "Validators can track the transaction history and link multiple addresses to a single user.", "isCorrect": false },
+          { "answerText": "Validators can track the transaction history and link multiple addresses to a single user.", "isCorrect": true },
```
```suggestion
          { "answerText": "Third-party services can track the transaction history and link multiple addresses to a single user.", "isCorrect": false },
```

NaitsabesMue replied:
Public auditability inherently makes address linking by third parties possible.

lucas-tortora commented on docs/site/static/json/about-iota/iota-architecture/iota-security.json:
```diff
@@ -4,9 +4,9 @@
         "answerOptions": [
           { "answerText": "Validators misbehaving during transaction processing.", "isCorrect": false },
-          { "answerText": "A flaw in the cryptographic methods securing private keys.", "isCorrect": false },
+          { "answerText": "A flaw in the cryptographic methods securing private keys.", "isCorrect": true },
           { "answerText": "Improper implementation of the asset logic in the smart contract by a third-party developer.", "isCorrect": true },
-          { "answerText": "Delegating asset access to an untrusted third-party service.", "isCorrect": false }
+          { "answerText": "Delegating asset access to an untrusted third-party service.", "isCorrect": true }
```
```suggestion
          { "answerText": "A flaw in the cryptographic methods securing private keys.", "isCorrect": false },
          { "answerText": "Improper implementation of the asset logic in the smart contract by a third-party developer.", "isCorrect": false },
          { "answerText": "Delegating asset access to an untrusted third-party service.", "isCorrect": false },
          { "answerText": "All of the above.", "isCorrect": true }
```

lucas-tortora reviewed (changes requested):
I'm afraid that the quizz component may not behave properly with multiple correct answers, but I think we can make it work with these suggestions

NaitsabesMue commented on docs/site/static/json/about-iota/iota-architecture/iota-security.json:
```diff
@@ -30,16 +30,16 @@
         "questionText": "Considering the public auditability of all transactions on IOTA, which of the following could be a potential vulnerability for users seeking privacy, even if they use pseudonymous addresses?",
         "answerOptions": [
-          { "answerText": "Validators can track the transaction history and link multiple addresses to a single user.", "isCorrect": false },
+          { "answerText": "Validators can track the transaction history and link multiple addresses to a single user.", "isCorrect": true },
```
```suggestion
          { "answerText": "Public auditability allows validators to modify past transactions.", "isCorrect": false },
```

NaitsabesMue committed:
Apply suggestions from code review

Proposed a new wrong answer, as third parties can track and link pseudonymous addresses
